spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultSubscriberFactory.java

```diff
@@ -162,6 +165,12 @@ public void setSubscriberStubRetrySettings(RetrySettings subscriberStubRetrySett
 		this.subscriberStubRetrySettings = subscriberStubRetrySettings;
 	}
 
+	public void setAcknowledger(PubSubAcknowledger acknowledger) {
```

Comment:
This is something wrong: it looks like an `acknowledger` abstraction is very tied with the `SubscriberStub`, which is produced exactly by this `SubscriberFactory`. With such a setter you don't give a protection that provided `PubSubAcknowledger` is not going to be based on some other subscriber instance.

Am I missing anything?

Reply:
You are correct ... to the `setAcknowledger` method will cause issues.  Coming to a concensus on the `boolean asyncAck` on the `PubSubSubscriberTemplate` negates this issue.

Reply:
I don't see that you have really resolved the issue: the setter is still here.
Am I missing anything else?

Thanks